src/ansi-c/c_typecheck_base.h

```diff
@@ -146,7 +146,16 @@ class c_typecheck_baset:
   virtual void typecheck_start_thread(codet &code);
 
   // contracts
+
+  /// \brief Typechecks a vector of assigns clause targets.
+  ///
+  /// \remark The vector will grow larger in case it contains conditional
+  /// targets of the form `c ? {t1, .. , tn}`, which get expanded to a
+  /// collection `{c ? t1, .. , c ? tn }` by distributing the condition `c`
+  /// on the list of targets.
   virtual void typecheck_spec_assigns(exprt::operandst &targets);
+  virtual void typecheck_spec_assigns_condition(exprt &condition);
+  virtual void typecheck_spec_assigns_target(exprt &target);
```

Comment:
I think you should use Doxygen groups or `\copydoc` to make clear which methods this comment refers to.

Reply:
it only concerns `typecheck_spec_assigns`, I added a new line to split the block of declarations in two.